### PR TITLE
String tensor support

### PR DESCRIPTION
**Pull request by GuyAv46**
For now, added only support for STRING tensors set with VALUE.

### CODE REVIEW

File: redisai/command_builder.py [hunk 1]
```diff
@@ -173,7 +173,10 @@ def tensorset(
         args = ["AI.TENSORSET", key, dtype, *shape, "BLOB", blob]
     elif isinstance(tensor, (list, tuple)):
         try:
-            dtype = utils.dtype_dict[dtype.lower()]
+            if np.dtype(dtype).num == np.dtype("str").num:
```
Comment (DvirDukhan) [inline redisai/command_builder.py:176]: please add a comment regarding `num` of `dtype`

File: redisai/utils.py [hunk 2]
```diff
@@ -15,20 +15,33 @@
...
 
 allowed_devices = {"CPU", "GPU"}
 allowed_backends = {"TF", "TFLITE", "TORCH", "ONNX"}
 
 
+def numpy_string2blob(tensor: np.ndarray) -> str:
+    """Convert the numpy string input from user to RedisAI blob."""
+    blob = ''
+    for i in range(tensor.size):
+        blob += tensor.flat[i] + "\0"
```
Comment (DvirDukhan) [inline redisai/utils.py:29]: `blob = "".join( [string + "\0" for string in tensor)])`

File: redisai/utils.py [hunk 3]
```diff
@@ -74,6 +89,21 @@ def recursive_bytetransform(arr: List[AnyStr], target: Callable) -> list:
     return arr
 
 
+def recursive_bytetransform_str(arr: List[AnyStr]) -> list:
```
Comment (DvirDukhan) [inline redisai/utils.py:92]: why the dedicated function?

**Review by DvirDukhan — COMMENTED** [review]
Looks good, few comments

File: redisai/utils.py [hunk 4]
```diff
@@ -38,7 +43,9 @@ def blob2numpy(
     """Convert `BLOB` result from RedisAI to `np.ndarray`."""
     mm = {"FLOAT": "float32", "DOUBLE": "float64"}
     dtype = mm.get(dtype, dtype.lower())
-    if mutable:
+    if dtype == 'string':
+        a = np.array(value.decode().split('\0')[:-1], dtype='str')
```
Comment (alonre24) [inline redisai/utils.py:47]: You take `[:-1]` because the last item is the empty string after the last `\0`in the blob right? Consider documenting it

**Review by alonre24 — APPROVED** [review]
Good job :)

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**